models/views/scratch/snowplow_unified_views_this_run.sql

```diff
@@ -115,24 +115,58 @@ with prep as (
 
     , coalesce(t.end_tstamp, p.derived_tstamp) as end_tstamp -- only page views with pings will have a row in table t
 
-    {% if var('snowplow__enable_web') %}
-      , coalesce(t.engaged_time_in_s, 0) as engaged_time_in_s -- where there are no pings, engaged time is 0.
-      , {{ datediff('p.derived_tstamp', 'coalesce(t.end_tstamp, p.derived_tstamp)', 'second') }} as absolute_time_in_s
-      , sd.hmax as horizontal_pixels_scrolled
-      , sd.vmax as vertical_pixels_scrolled
-      , sd.relative_hmax as horizontal_percentage_scrolled
-      , sd.relative_vmax as vertical_percentage_scrolled
+    {% if var('snowplow__enable_web') or var('snowplow__enable_screen_summary_context', false) %}
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.foreground_sec,{% endif %}
+        t.engaged_time_in_s,
+        0 -- where there are no pings, engaged time is 0.
+      ) as engaged_time_in_s
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.foreground_sec + coalesce(ss.background_sec, 0),{% endif %}
+        {{ datediff('p.derived_tstamp', 'coalesce(t.end_tstamp, p.derived_tstamp)', 'second') }},
+        0
+      ) as absolute_time_in_s
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.max_x_offset,{% endif %}
+        sd.hmax,
+        null
+      ) as horizontal_pixels_scrolled
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.max_y_offset,{% endif %}
+        sd.vmax,
+        null
+      ) as vertical_pixels_scrolled
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.horizontal_percentage_scrolled,{% endif %}
+        sd.relative_hmax,
+        null
+      ) as horizontal_percentage_scrolled
+      , coalesce(
+        {% if var('snowplow__enable_screen_summary_context', false) %}ss.vertical_percentage_scrolled,{% endif %}
+        sd.relative_vmax,
+        null
+      ) as vertical_percentage_scrolled
+    {% endif %}
+    {% if var('snowplow__enable_screen_summary_context', false) %}
+      , ss.last_item_index as last_list_item_index
+      , ss.items_count as list_items_count
+      , ss.list_items_percentage_scrolled
     {% endif %}
 
     , {{ snowplow_utils.current_timestamp_in_utc() }} as model_tstamp
 
   from prep p
 
   left join {{ ref('snowplow_unified_pv_engaged_time') }} t
-  on p.view_id = t.view_id and p.session_identifier = t.session_identifier
+  on p.platform = 'web' and p.view_id = t.view_id and p.session_identifier = t.session_identifier
```

Comment:
Dumb question, how confident are we that platform is 100% correct?

Reply:
we are already relying on this in `users_sessions_this_run`, if this needs case or null handling, let's amend it there too

Reply:
All events should have a platform, it's set by default on the tracker, although the user may decide to change it.

I only added it here for performance reasons so that we don't join on the web and mobile tables unnecessarily. I don't mind removing it if we want to be more careful.

One thing to note: the mobile data model can be used also on platforms other than `mob` – on `tv`, `desktop`, ... That's why I use `platform <> 'web'` here for mobile, maybe we could do the same in the `users_sessions_this_run`.